### ISSUE TITLE
Test TestInspectApiBridgeNetworkSettings12{0,1} can fail if container is already stopped before inspect

### ISSUE THREAD

**Issue by Snorch** [docker/docker#17646]
I found that TestInspectApiBridgeNetworkSettings12{0,1} fails intermitently with v1.9.0-rc4

Container is started and then we check inspect, but what if inspect proceed after container stop? Output will have empty IPAddress == "". One can add sleep to reproduce it better(100%):
```
@@ -156,6 +157,8 @@ func (s *DockerSuite) TestInspectApiEmptyFieldsInConfigPre121(c *check.C) {
 func (s *DockerSuite) TestInspectApiBridgeNetworkSettings120(c *check.C) {
        out, _ := dockerCmd(c, "run", "-d", "busybox", "true")
 
+       time.Sleep(time.Second * 20)
+
        cleanedContainerID := strings.TrimSpace(out)
        body := getInspectBody(c, "v1.20", cleanedContainerID)
 
@@ -170,6 +173,7 @@ func (s *DockerSuite) TestInspectApiBridgeNetworkSettings120(c *check.C) {
 func (s *DockerSuite) TestInspectApiBridgeNetworkSettings121(c *check.C) {
        out, _ := dockerCmd(c, "run", "-d", "busybox", "true")
        cleanedContainerID := strings.TrimSpace(out)
+       time.Sleep(time.Second * 20)
 
        body := getInspectBody(c, "v1.21", cleanedContainerID)
```
Resulting output of tests:
```
docker run --rm -it --privileged -e BUILDFLAGS -e DOCKER_CLIENTONLY -e DOCKER_DEBUG -e DOCKER_EXECDRIVER -e DOCKER_EXPERIMENTAL -e DOCKER_REMAP_ROOT -e DOCKER_GRAPHDRIVER -e DOCKER_STORAGE_OPTS -e DOCKER_USERLANDPROXY -e TESTDIRS -e TESTFLAGS -e TIMEOUT -v "/home/snorch/docker/docker/bundles:/go/src/github.com/docker/docker/bundles" "docker-dev:HEAD" hack/make.sh binary test-integration-cli

bundles/1.9.0-rc4 already exists. Removing.

---> Making bundle: binary (in bundles/1.9.0-rc4/binary)
Building: bundles/1.9.0-rc4/binary/docker-1.9.0-rc4
Created binary: bundles/1.9.0-rc4/binary/docker-1.9.0-rc4

---> Making bundle: test-integration-cli (in bundles/1.9.0-rc4/test-integration-cli)
---> Making bundle: .integration-daemon-start (in bundles/1.9.0-rc4/test-integration-cli)
+++ exec docker daemon --debug --host unix:///go/src/github.com/docker/docker/bundles/1.9.0-rc4/test-integration-cli/docker.sock --storage-driver vfs --exec-driver native --pidfile bundles/1.9.0-rc4/test-integration-cli/docker.pid --userland-proxy=true
---> Making bundle: .integration-daemon-setup (in bundles/1.9.0-rc4/test-integration-cli)
---> Making bundle: .ensure-emptyfs (in bundles/1.9.0-rc4/test-integration-cli)
++++ tar -cC bundles/1.9.0-rc4/test-integration-cli/emptyfs .
++++ docker load
---> Making bundle: .ensure-frozen-images (in bundles/1.9.0-rc4/test-integration-cli)
++++ tar -cC /docker-frozen-images .
++++ docker load
---> Making bundle: .ensure-httpserver (in bundles/1.9.0-rc4/test-integration-cli)
+ go test -race -check.f DockerSuite.TestInspectApiBridgeNetworkSettings12* -test.timeout=60m -check.v github.com/docker/docker/integration-cli
INFO: Testing against a local daemon

----------------------------------------------------------------------
FAIL: docker_api_inspect_test.go:157: DockerSuite.TestInspectApiBridgeNetworkSettings120

docker_api_inspect_test.go:170:
    c.Assert(settings.IPAddress, checker.Not(checker.HasLen), 0)
... obtained string = ""
... n int = 0


----------------------------------------------------------------------
FAIL: docker_api_inspect_test.go:173: DockerSuite.TestInspectApiBridgeNetworkSettings121

docker_api_inspect_test.go:185:
    c.Assert(settings.IPAddress, checker.Not(checker.HasLen), 0)
... obtained string = ""
... n int = 0

OOPS: 0 passed, 2 FAILED
--- FAIL: Test (42.19s)
FAIL
coverage: 11.1% of statements
exit status 1
FAIL	github.com/docker/docker/integration-cli	42.241s
---> Making bundle: .integration-daemon-stop (in bundles/1.9.0-rc4/test-integration-cli)
+++++ cat bundles/1.9.0-rc4/test-integration-cli/docker.pid
++++ kill 874
make: *** [test-integration-cli] Error 1
```
Other info:
```
sudo docker info
Containers: 12
Images: 462
Server Version: 1.9.0-rc4
Storage Driver: devicemapper
 Pool Name: docker-253:1-1182990-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: extfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 31.52 GB
 Data Space Total: 107.4 GB
 Data Space Available: 5.326 GB
 Metadata Space Used: 39.81 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.108 GB
 Udev Sync Supported: false
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.82-git (2013-10-04)
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.19.8-100.fc20.x86_64
Operating System: Fedora 20 (Heisenbug)
CPUs: 4
Total Memory: 15.55 GiB
Name: dhcp-10-30-28-101.sw.ru
ID: Q3EP:SQO6:I3JK:PDAS:2QAY:GWEQ:M7ZN:CEEF:OBZO:QEHR:HVJ3:ETAR
```

**Comment by cpuguy83** [comment]
Thanks, fixing.

**Comment by thaJeztah** [comment]
ping @cpuguy83 were you working on this?

**Comment by cpuguy83** [comment]
Yes, this was fixed in #17659
Forgot to add the `fixes #17659` to the PR.

**Comment by thaJeztah** [comment]
Great! Thanks!
